honeybadgermpc/ntl/rsdecode_impl.h

```diff
@@ -401,4 +407,110 @@ bool gao_interpolate_fft(vec_ZZ_p &res_vec, vec_ZZ_p &err_vec,
     }
 
     return true;
-}
\ No newline at end of file
+}
+
+typedef std::string OpaqueZZp;
+
+void printLimbs(vector<ZZ_limb_t> &l) {
+    long size = l.size();
+    for (int i =0; i< size; i++) {
+        cout << l[i] << endl;
+    }
+}
+
+void VecLimbToZZ_p(ZZ_p &zzp, vector<ZZ_limb_t> &arr) {
+    long size = arr.size();    
+    ZZ_limb_t limbs[size];
+    for (int i =0; i< size; i++) {
+        limbs[i] = arr[i];
+    }
+    ZZ a;
+    ZZ_limb_t *const l = limbs;
+    ZZ_limbs_set(a, l, size);
+    zzp =  to_ZZ_p(a);    
+}
+
+void ZZ_pToVecLimb(vector<ZZ_limb_t> &limbs, ZZ_p &x) {
+    ZZ zz = rep(x);
+    long size = zz.size();
+    limbs.reserve(size);
+    const ZZ_limb_t *l = ZZ_limbs_get(zz);
+    for (int i =0; i< size; i++) {
+        limbs.push_back(l[i]);
+    }
+}
+
+class ZZplimbs {
+   public:
+        long size;
+        ZZ_limb_t *l;
+
+        template <class B>
+        void serialize(B& buf) const {
+            buf << size;
+            for(int i =0; i<size; i++){
+                buf << l[i];
+            }
+        }
+
+        template <class B>
+        void parse(B& buf) {
+            buf >> size;
+            l = new ZZ_limb_t[size];
```

Comment:
There should be an extra check for size here (size <= 32?). Also, is there a way to check if HPS serialization failed for any reason?

Reply:
Meant to review #353. You can ignore this.

Reply:
great point, i'd want to be able to catch (int python) any exception due to serialization failure